bndtools.core/src/org/bndtools/core/editors/ImportPackageQuickFixProcessor.java

```diff
@@ -0,0 +1,578 @@
+        }
+        while (selectedNode instanceof Name) {
+            selectedNode = selectedNode.getParent();
+        }
+        Name name = null;
+        if (selectedNode instanceof SimpleType) {
+            name = ((SimpleType) selectedNode).getName();
+            if (!name.isQualifiedName()) {
+                return null;
+            }
+            name = ((QualifiedName) name).getQualifier();
+        } else if (selectedNode instanceof NameQualifiedType) {
+            name = ((NameQualifiedType) selectedNode).getQualifier();
+        }
+
+        if (name == null) {
+            return null;
+        }
+
+        return skipClasses(name);
+    }
+
+    @Override
+    public IJavaCompletionProposal[] getCorrections(IInvocationContext context, IProblemLocation[] locations) throws CoreException {
+        Set<String> pkgs = new HashSet<>(locations.length * 2 + 1);
+
+        for (IProblemLocation location : locations) {
+            Name name;
+            switch (location.getProblemId()) {
+                case IProblem.ImportNotFound :
+                    name = getPackageFromImportNotFound(context, location);
+                    break;
+                case IProblem.UndefinedType :
+                    name = getPackageFromUndefinedType(context, location);
+                    break;
+                default :
+                    continue;
+            }
+
+            if (name == null) {
+                continue;
+            }
+
+            final String pkg = name.getFullyQualifiedName();
+            // Don't suggest adding a bundle to fix missing package in import if current Compilation Unit
+            // is already part of that package.
+            final PackageDeclaration ourPkg = context.getASTRoot()
+                .getPackage();
+            if (ourPkg != null && pkg.equals(ourPkg.getName()
+                .getFullyQualifiedName())) {
+                continue;
+            }
+            pkgs.add(pkg);
+        }
+        if (pkgs.isEmpty()) {
+            return null;
+        }
+        return getSuggestions(pkgs, context);
+    }
+
+    /**
+     * Converts a package capability into a bundle symbolic name.
+     *
+     * @param cap The capability object describing the package being imported.
+     * @return A user-friendly string with the BSN.
+     */
+    public static String capabilityToBSN(Capability cap) {
+        List<Capability> caps = cap.getResource()
+            .getCapabilities(BundleNamespace.BUNDLE_NAMESPACE);
+        cap = caps.get(0);
+        final Map<String, Object> attributes = cap.getAttributes();
+        return attributes.get(BundleNamespace.BUNDLE_NAMESPACE) + "";
+    }
+
+    /**
+     * Converts a package capability into a VersionedClause.
+     *
+     * @param cap The capability object describing the package being imported.
+     * @return A user-friendly string with the BSN and version.
+     */
+    public static VersionedClause capabilityToVersion(Capability cap) {
```

Comment:
This method does not appear to be used.

Reply:
It was used in an earlier iteration, when I was toying with the idea of showing suggestions for individual available bundle versions, rather than simply suggesting the bare bsn. I left the code in in case I decided to add this functionality again in the future (similarly with `BndBuildPathHandler.containsBundle(VersionedClause)`). I removed it now though - it won't be hard to add again if needed.